other/docker/misra/Makefile

```diff
@@ -0,0 +1,166 @@
+# The lowercase character 'l' shall not be used in a literal suffix.
+#
+# Reason: False positive. We don't use 'l', but this flags 'ulOutBufLen'.
+SUPPRESSIONS += 7.3
+# Operands shall not be of an inappropriate essential type.
+#
+# Reason: This diagnoses (1 << n) and wants us to use (1u << n). That's fair,
+# but this diagnostic is impossible to fix for ((1u << n) >> m).
+SUPPRESSIONS += 10.1
+# Both operands of an operator in which the usual arithmetic conversions are performed shall have the same essential type category.
+#
+# Reason: This warns about ((unsigned)n == 0) and other constant comparisons.
+SUPPRESSIONS += 10.4
+# The value of a composite expression shall not be cast to a different essential type category or a wider essential type.
+#
+# TODO(iphydf): investigate.
+SUPPRESSIONS += 10.8
+# A conversion should not be performed from pointer to void into pointer to object.
+#
+# Reason: this is needed for generic callbacks to make any sense.
+SUPPRESSIONS += 11.5
+# The precedence of operators within expressions should be made explicit.
+#
+# Reason: this asks us to add a lot of extra parentheses that don't really help
+# readability. We expect people to know basic precedence. GCC has a better
+# diagnostic requiring parentheses around less common operators.
+SUPPRESSIONS += 12.1
+# The comma operator should not be used.
+#
+# Reason: We don't use the comma operator (cimple doesn't even parse it). This is
+# all false positives.
+SUPPRESSIONS += 12.3
+# Evaluation of constant expressions should not lead to unsigned integer wrap-around.
+#
+# Reason: False positives on UINT64_MAX.
+SUPPRESSIONS += 12.4
+# A full expression containing an increment (++) or decrement (--) operator should have no other potential side effects other than that caused by the increment or decrement operator.
+#
+# TODO(iphydf): maybe fix?
+SUPPRESSIONS += 13.3
+# The controlling expression of an if statement and the controlling expression of an iteration-statement shall have essentially Boolean type.
+#
+# Reason: We actually follow this rule, but cppcheck's implementation is flawed and has false positives.
+SUPPRESSIONS += 14.4
+# The goto statement should not be used.
+#
+# TODO(iphydf): Get rid of goto.
+SUPPRESSIONS += 15.1
+# A function should have a single point of exit at the end.
+#
+# Reason: This doesn't make code much clearer. Sometimes this is useful for
+# putting all the cleanup code in one spot, but often an early return improves
+# readability.
+SUPPRESSIONS += 15.5
+# All if . . else if constructs shall be terminated with an else statement.
+#
+# TODO(iphydf): Why is this a good idea?
+SUPPRESSIONS += 15.7
+# An unconditional break statement shall terminate every switch-clause.
+#
+# Reason: This conflicts with "break unused after abort()". MISRA doesn't allow
+# abort(), but we use it, so this rule must be disabled, too.
+SUPPRESSIONS += 16.3
+# Every switch statement shall have a default label.
+#
+# TODO(iphydf): maybe fix?
```

Comment:
IIRC default may conflict with a diagnostic from compilers that check that alll enum values are covered. I don't think that should be fixed by MISRA rules, but rather by using that compiler diagnostic.

Reply:
Yeah, I agree. Removed the TODO.